backend/index.ts

```diff
@@ -18,31 +18,162 @@ delete --> delete
+
+//Creates A User Object
+app.post('/user', (req, res) => {
+  const {mail, password, username, profilePicture,} = req.body;
+  // res -> status code
+});
 
-// Returns user data (including communities they're subbed to)
+//Returns user data
 app.get('/user', (req, res) => {
-  // this will determine the offset of how many posts to give the user
-  // we are NOT making it unique to each user, it's ok.
   const {userID} = req.query; 
-  res.json({
-    0: ["25 post objects"]
-  })
+  //res -> json object
 });
 
 //Adds a community to the list of communities users are in
 app.put('/subscribeToCommunity', (req, res) => {
-  // do this things with that
   const {userID, communityID} = req.body
   // res -> status code
 });
 
+//Changes the profile picture of the user once profile has already been created
 app.put('/changeProfilePic', (req, res) => {
   // do this things with that
   const {userID, photoURL} = req.body
   // res -> status code
 });
 
+//Deletes User
+app.delete('/user', (req, res) => {
+  const {userID} = req.body
+  // res -> status code
+});
+
+//////////////////////////////////////////////////////////////////////////////////////COMMUNITY\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\
+
+//Create A Community Object
+app.post('/community', (req, res) => {
+  const {userID, communityName, description, picture} = req.body;
+  // res -> status code
+});
+
+//Gets the all the data from a community from the database
+app.get('/community', (req, res) => {
+  const {communityID} = req.query; 
+  // res -> json object
+});
+
+//Allows moderators to edit description of community
+app.put('/editDescription', (req, res) => {
+  const {userID, communityID, description} = req.query; 
+  // res -> status code
+});
+
+//Changes the profile picture of the community
+app.put('/changeCommunityPic', (req, res) => {
+  // do this things with that
+  const {userID, communityID, photoURL} = req.body
+  // res -> status code
+});
+
+//Gives user moderator
+app.put('/makeMod', (req, res) => {
+  // do this things with that
+  const {communityID, userID} = req.body
+  // res -> status code
+});
+
+//Bans User From Community
+app.delete('/banUser', (req, res) => {
+  // do this things with that
+  const {communityID, userID} = req.body
+  // res -> status code
+});
+
+//Deletes Community
+app.delete('/user', (req, res) => {
+  const {userID, communityID} = req.body
+  // res -> status code
+});
+
+//////////////////////////////////////////////////////////////////////////////////////COMMUNITY\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\
+
+//Creates A Post
+app.post('/post', (req, res) => {
+  const {userID, title, type, body, tags} = req.body;
+  // res -> status code
+});
+
+//Gets Post Data
+app.get('/post', (req, res) => {
+  const {userID, postID} = req.query;
+  // res -> JSON object
+});
+
+//Edit Body Of Post
+app.put('/post', (req, res) => {
```

Comment:
For convention, we should probably change edit to date. I think even for something like the description of a post, or the photo of a user, everything should be changed to data so it's easier to deal with in the long run.